presto-native-execution/presto_cpp/main/TaskResource.h

```diff
@@ -15,18 +15,21 @@
 
 #include "presto_cpp/main/TaskManager.h"
 #include "presto_cpp/main/http/HttpServer.h"
+#include "presto_cpp/main/types/PrestoToVeloxPlanValidator.h"
 #include "velox/common/memory/Memory.h"
 
 namespace facebook::presto {
 
 class TaskResource {
  public:
   explicit TaskResource(
-      TaskManager& taskManager,
       velox::memory::MemoryPool* pool,
-      folly::Executor* httpSrvCpuExecutor)
+      folly::Executor* httpSrvCpuExecutor,
+      PrestoToVeloxPlanValidator* planValidator,
```

Comment:
usually a good idea to pass the shared_ptr so that lifecycle and ownership are more explicitly defined.

Reply:
(or unique_ptr, depending on how you organize things).